src/util.rs

```diff
@@ -35,10 +36,15 @@ pub fn glob_match_file(mut path: PathBuf) -> Result<PathBuf, ActixError> {
             )
             .to_string(),
     );
-    if let Some(glob_path) = glob(&format!("{}.[0-9]*", path.to_string_lossy()))
-        .map_err(error::ErrorInternalServerError)?
-        .last()
-    {
+
+    let path_string = path.to_string_lossy().into_owned();
+    let glob_match = match spawn_blocking(move || glob(&format!("{}.[0-9]*", path_string))).await {
+        Ok(Ok(m)) => m.last(),
+        Ok(Err(e)) => return Err(error::ErrorInternalServerError(e)),
+        Err(e) => return Err(error::ErrorInternalServerError(e)),
+    };
```

Comment:
Why spawn a task?

Reply:
`glob` is sync only

Reply:
Out of curiosity, isn't spawning more resource intensive than the glob call + processing? Furthermore, it is spawned as blocking, so how does this help the event loop?
Maybe I miss something here.

Reply:
`spawn_blocking` blocks this task and waits until it gets a result. The `await` yields control back to the event loop to allow it to process other tasks (such as other requests).

I strongly suspect that spawning a thread is significantly faster than multiple filesystem hits.